packages/build-utils/src/convert-runtime-to-plugin.ts

```diff
@@ -0,0 +1,85 @@
+import fs from 'fs-extra';
+import { join, dirname, relative } from 'path';
+import glob from './fs/glob';
+import { normalizePath } from './fs/normalize-path';
+import { FILES_SYMBOL, Lambda } from './lambda';
+import type FileBlob from './file-blob';
+import type { BuildOptions, Files } from './types';
+
+const OUTPUT_DIR = '.output';
+
+/**
+ *
+ * @param buildRuntime a legacy build() function from a Runtime
+ * @param ext the file extension, for example `.py`
+ */
+export async function convertRuntimeToPlugin(
+  buildRuntime: (options: BuildOptions) => Promise<{ output: Lambda }>,
+  ext: string
+) {
+  return async function build({ workPath }: { workPath: string }) {
+    const opts = { cwd: workPath };
+    const files = await glob('**', opts);
+    const entrypoints = await glob(`api/**/*${ext}`, opts);
+    const dir = join(workPath, OUTPUT_DIR, 'inputs', 'runtime-temp');
+    await fs.ensureDir(dir);
+    for (const entrypoint of Object.keys(entrypoints)) {
+      const { output } = await buildRuntime({
+        files,
+        entrypoint,
+        workPath,
+        // TODO: What about includeFiles/excludeFiles?
```

Comment:
I don't think we can consider it in this case.

With the new system we'd expect the build process to take care of that, while the old system expected it to be there upfront.

Maybe we'd need to read `vercel.json` 🤔

Reply:
I just realized there won't even be a vercel.json config to read because it used to read `functions` or `builds` but those won't exist for plugins.

And I think the plan is for legacy builders/runtimes to provide a new npm package so users would have to opt into the new plugin and thus we could change the way this works: perhaps the plugin could listen for environment variables?

Reply:
I updated the code to read `vercel.json` and handle functions in f0d8d3d